src/ceph-daemon

```diff
@@ -0,0 +1,924 @@
+RestartSec=10s
+TimeoutStartSec=120
+TimeoutStopSec=15
+StartLimitInterval=30min
+StartLimitBurst=5
+
+[Install]
+WantedBy=ceph-{fsid}.target
+""".format(fsid=fsid, data_dir=args.data_dir)
+    return u
+
+def gen_ssh_key(fsid):
+    tmp_dir = tempfile.TemporaryDirectory()
+    path = tmp_dir.name + '/key'
+    subprocess.check_output([
+        'ssh-keygen',
+        '-C', 'ceph-%s' % fsid,
+        '-N', '',
+        '-f', path
+    ])
+    with open(path, 'r') as f:
+        secret = f.read()
+    with open(path + '.pub', 'r') as f:
+        pub = f.read()
+    os.unlink(path)
+    os.unlink(path + '.pub')
+    tmp_dir.cleanup()
+    return (secret, pub)
+
+##################################
+
+class CephContainer:
+    def __init__(self,
+                 image,
+                 entrypoint,
+                 args=[],
+                 volume_mounts={},
+                 cname='',
+                 dname='',
+                 podman_args=[]):
+        self.image = image
+        self.entrypoint = entrypoint
+        self.args = args
+        self.volume_mounts = volume_mounts
+        self.cname = cname
+        self.dname = dname
+        if dname:
+            (self.daemon_type, self.daemon_id) = dname.split('.')
+        else:
+            self.daemon_type = None
+            self.daemon_id = None
+        self.podman_args = podman_args
+
+    def run_cmd(self):
+        vols = sum(
+            [['-v', f'{host_dir}:{container_dir}']
+             for host_dir, container_dir in self.volume_mounts.items()], [])
+        envs = [
+            '-e', f'CONTAINER_IMAGE={self.image}',
+            '-e', f'NODE_NAME={get_hostname()}',
+        ]
+        cname = ['--name', self.cname] if self.cname else []
+        return [
+            find_program('podman'),
+            'run',
+            '--rm',
+            '--net=host',
+        ] + self.podman_args + cname + envs + vols + [
+            '--entrypoint', f'/usr/bin/{self.entrypoint}',
+            self.image
+        ] + self.args
+
+    def shell_cmd(self):
+        vols = sum(
+            [['-v', f'{host_dir}:{container_dir}']
+             for host_dir, container_dir in self.volume_mounts.items()], [])
+        envs = [
+            '-e', f'CONTAINER_IMAGE={self.image}',
+            '-e', f'NODE_NAME={get_hostname()}',
+        ]
+        return [
+            find_program('podman'),
+            'run',
+            '-it',
+            '--net=host',
+        ] + self.podman_args + envs + vols + [
+            '--entrypoint', '/bin/bash',
+            self.image
+        ]
+
+    def exec_cmd(self, cmd):
+        return [
+            find_program('podman'),
+            'exec',
+            '-it',
+            self.cname,
+        ] + cmd
+
+    def run(self):
+        logging.debug(self.run_cmd())
+        return subprocess.check_output(self.run_cmd())
+
+##################################
+
+def command_version():
+    out = CephContainer(args.image, 'ceph', ['--version']).run()
```

Comment:
the way you would do this in remoto is:

```
>>> from remoto.process import check
>>> from remoto import connection
>>> Connection = connection.get('podman')
>>> conn = Connection(hostname='localhost', container_id=args.image))
>>> out, err, exit_code = check(conn, ['ceph', '--version'])
>>> return out[0]
```

This is an example usage with a helper function that takes the connection as an argument (code lifted from ceph-medic):

```python
def ceph_version(conn):
    try:
        output, _, exit_code = check(conn, ['ceph', '--version'])
        if exit_code != 0:
            conn.logger.error('Non zero exit status received, unable to retrieve information')
            return
        return output[0]
    except RuntimeError:
        conn.logger.exception('failed to fetch ceph version')
```